packages/editor/src/browser/breadcrumbs/breadcrumbs-utils.ts

```diff
@@ -0,0 +1,96 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Breadcrumbs } from './breadcrumbs';
+import { OutlineSymbolInformationNode } from '@theia/outline-view/lib/browser';
+
+/**
+ * Traverse upstream (starting with the HTML element `child`) to find a parent HTML element
+ * that has the CSS class `Breadcrumbs.Styles.BREADCRUMB_ITEM`.
+ */
+export function findParentItemHtmlElement(child: HTMLElement): HTMLElement | undefined {
```

Comment:
please it a class, see https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#di-function-export